### PR TITLE
Guaranteeing that Flex is installed THEN fwbundle

### PR DESCRIPTION
**Pull request by weaverryan**
See #238.

I talked with @nicolas-grekas about this. By merging the Flex recipe into FWBundle, we created a possibility (which I hit while creating our upgrade tutorial) that the `.env.dist` file is not present. And then, recipes may start installing incorrectly because this file is needed (so that recipes can update it and `.env`).

The fix is this PR + to revert https://github.com/symfony/recipes/pull/295

### CODE REVIEW

File: src/Flex.php [hunk 1]
```diff
@@ -118,8 +118,20 @@ public function record(PackageEvent $event)
     {
         $operation = $event->getOperation();
         if ($this->shouldRecordOperation($operation)) {
-            if ($operation instanceof InstallOperation && 'symfony/framework-bundle' === $operation->getPackage()->getName()) {
-                array_unshift($this->operations, $operation);
+            if ($operation instanceof InstallOperation && in_array($packageName = $operation->getPackage()->getName(), ['symfony/framework-bundle', 'symfony/flex'])) {
+                if ($packageName === 'symfony/flex') {
```
Comment (fabpot) [inline src/Flex.php]: Yoda condition?

File: src/Flex.php [hunk 2]
```diff
@@ -118,8 +118,20 @@ public function record(PackageEvent $event)
...
+                    array_unshift($this->operations, $operation);
+                } else {
+                    if ($this->operations && $this->operations[0] instanceof InstallOperation && $this->operations[0]->getPackage()->getName() === 'symfony/flex') {
```
Comment (fabpot) [inline src/Flex.php]: Yoda here as well

Comment (weaverryan) [reply src/Flex.php]: Right you are ;)

**Comment by fabpot** [comment]
Thank you @weaverryan.

**Pull request merged**